nodebuilder/pruner/module.go

```diff
@@ -0,0 +1,75 @@
+package pruner
+
+import (
+	"context"
+
+	"go.uber.org/fx"
+
+	"github.com/celestiaorg/celestia-node/nodebuilder/node"
+	"github.com/celestiaorg/celestia-node/pruner"
+	"github.com/celestiaorg/celestia-node/pruner/archival"
+	"github.com/celestiaorg/celestia-node/pruner/full"
+	"github.com/celestiaorg/celestia-node/pruner/light"
+	"github.com/celestiaorg/celestia-node/share/eds"
+)
+
+func ConstructModule(tp node.Type, cfg *Config) fx.Option {
+	if !cfg.EnableService {
+		switch tp {
+		case node.Light:
+			// light nodes are still subject to sampling within window
+			// even if pruning is not enabled.
+			return fx.Supply(light.Window)
+		case node.Full, node.Bridge:
+			return fx.Supply(archival.Window)
+		default:
+			panic("unknown node type")
+		}
+	}
+
+	baseComponents := fx.Options(
+		fx.Provide(fx.Annotate(
+			newPrunerService,
+			fx.OnStart(func(ctx context.Context, p *pruner.Service) error {
+				return p.Start(ctx)
+			}),
+			fx.OnStop(func(ctx context.Context, p *pruner.Service) error {
+				return p.Stop(ctx)
+			}),
+		)),
+		// This is necessary to invoke the pruner service as independent thanks to a
+		// quirk in FX.
+		fx.Invoke(func(p *pruner.Service) {}),
+	)
+
+	switch tp {
+	case node.Full:
+		return fx.Module("prune",
+			baseComponents,
+			fx.Provide(func(store *eds.Store) pruner.Pruner {
+				return full.NewPruner(store)
+			}),
+			fx.Supply(full.Window),
+		)
+	case node.Bridge:
+		return fx.Module("prune",
+			baseComponents,
+			fx.Provide(func(store *eds.Store) pruner.Pruner {
+				return full.NewPruner(store)
+			}),
+			fx.Supply(full.Window),
+		)
```

Comment:
Archival sync PR will make it so that they differ, that's why I had them separated but I guess I could unify now if you think it's super important for this PR

Reply:
Doesn't matter then. Didn't know those will need to be separated.